### PR TITLE
Switch to using opencontainers/selinux for selinux bindings

### PR DESCRIPTION
**Pull request by runcom**
Carry of https://github.com/docker/docker/pull/32011, hopefully tests are green here.

@cpuguy83 @thaJeztah @vdemeester @rhatdan PTAL

Signed-off-by: Antonio Murdaca <runcom@redhat.com>

### CODE REVIEW

**Comment by runcom** [comment]
Ping anyone, why aren't tests running for this PR?

File: vendor/github.com/opencontainers/selinux/go-selinux/selinux.go [hunk 1]
```diff
@@ -166,43 +208,55 @@ func readCon(name string) (string, error) {
...
-	return system.Lsetxattr(path, xattrNameSelinux, []byte(scon), 0)
+// SetFileLabel sets the SELinux label for this path or returns an error.
+func SetFileLabel(path string, label string) error {
+	return lsetxattr(path, xattrNameSelinux, []byte(label), 0)
 }
 
-// Getfilecon returns the SELinux label for this path or returns an error.
-func Getfilecon(path string) (string, error) {
-	con, err := system.Lgetxattr(path, xattrNameSelinux)
+// Filecon returns the SELinux label for this path or returns an error.
```
Comment (TomSweeneyRedHat) [inline vendor/github.com/opencontainers/selinux/go-selinux/selinux.go]: Filecon to FileLabel in this comment.

Comment (runcom) [inline vendor/github.com/opencontainers/selinux/go-selinux/selinux.go]: Nice catch, could you open a PR to fix this in opencontainers/selinux?

File: vendor/github.com/opencontainers/selinux/go-selinux/selinux.go [hunk 2]
```diff
@@ -498,14 +580,14 @@ func DupSecOpt(src string) []string {
 		con["level"] == "" {
 		return nil
 	}
-	return []string{"label=user:" + con["user"],
-		"label=role:" + con["role"],
```
Comment (cpuguy83) [inline vendor/github.com/opencontainers/selinux/go-selinux/selinux.go]: This change seems like it's going to break stuff.
Why the change?

Definitely breaking https://github.com/moby/moby/blob/master/daemon/cluster/executor/container/container.go#L624

**Review by cpuguy83 — CHANGES_REQUESTED** [review]
I don't think we should make any changes until we have a strongly-typed object for handing off selinux context from the daemon.

For instance in swarm we use this: https://github.com/moby/moby/blob/091b5e68ea735bf4e8ece708bbc8c413a32eab73/api/types/swarm/container.go#L25

File: vendor/github.com/opencontainers/selinux/go-selinux/selinux.go (hunk 2 above)
Comment (runcom) [inline vendor/github.com/opencontainers/selinux/go-selinux/selinux.go]: @cpuguy83 selinux hasn't any notion of `label=` prefixed labels. The label is just the label, not `label=label` <- this is something docker did afacit, nothing related to selinux /cc @rhatdan

**Comment by runcom** [comment]
> I don't think we should make any changes until we have a strongly-typed object for handing off selinux context from the daemon.

I don't understand....this change is super-transparent and what you're saying can be postponed for sure, in the mean time we'll switch to OCI/selinux since there's no libcontainer/selinux anymore. If something happen to be incorrect we'll fix OCI/selinux directly that way...

File: vendor/github.com/opencontainers/selinux/go-selinux/selinux.go (hunk 2 above)
Comment (rhatdan) [inline vendor/github.com/opencontainers/selinux/go-selinux/selinux.go]: label should not be in any SELinux specific section. this should be in the label section but not in the selinux specific section.

**Comment by cpuguy83** [comment]
Ok I see you've got this label conversion happening in Docker now. That's fine.
I'll open an issue to discuss a strong type that we can convert to to make sure this kind of stuff is checked at compile time.

Will need to do some manual testing on this just to make sure we're not breaking anything (seems ok, but just want to be sure).

**Review by cpuguy83 — APPROVED** [review]
LGTM

**Comment by runcom** [comment]
everything green :green_heart:

**Review by vdemeester — APPROVED** [review]
LGTM 👍

**Pull request merged**